src/abstractdataframe/io.jl

```diff
@@ -164,128 +165,150 @@ function html_escape(cell::AbstractString)
     _check_consistency(df)
 
-    # we will pass around this buffer to avoid its reallocation in ourstrwidth
-    buffer = IOBuffer(Vector{UInt8}(undef, 80), read=true, write=true)
+    names_str = names(df)
+    types = Any[eltype(c) for c in eachcol(df)]
+    types_str = batch_compacttype(types, 9)
+    types_str_complete = batch_compacttype(types, 256)
 
-    if rowid !== nothing
-        if size(df, 2) == 0
-            rowid = nothing
-        elseif size(df, 1) != 1
-            throw(ArgumentError("rowid may be passed only with a single row data frame"))
+    # For consistency, if `kwargs` has `compact_printng`, we must use it.
+    compact_printing::Bool = get(kwargs, :compact_printing, get(io, :compact, true))
+
+    num_rows, num_cols = size(df)
+
+    # By default, we align the columns to the left unless they are numbers,
+    # which is checked in the following.
+    alignment = fill(:l, num_cols)
+
+    for i = 1:num_cols
+        type_i = nonmissingtype(types[i])
+
+        if type_i <: Number
+            alignment[i] = :r
         end
     end
 
-    mxrow, mxcol = size(df)
     if get(io, :limit, false)
-        tty_rows, tty_cols = displaysize(io)
-        mxrow = min(mxrow, tty_rows)
-        maxwidths = getmaxwidths(df, io, 1:mxrow, 0:-1, :X, nothing, true, buffer, 0) .+ 2
-        mxcol = min(mxcol, searchsortedfirst(cumsum(maxwidths), tty_cols))
-    end
+        # Obtain the maximum number of rows and columns that we can print from
+        # environments variables.
+        mxrow = tryparse(Int, get(ENV, "DATAFRAMES_ROWS", "25"))
 
-    cnames = _names(df)[1:mxcol]
-    write(io, "<div class=\"data-frame\">")
-    if summary
-        write(io, "<p>$(digitsep(nrow(df))) rows × $(digitsep(ncol(df))) columns")
-        if mxcol < size(df, 2)
-            write(io, " (omitted printing of $(size(df, 2)-mxcol) columns)")
+        if isnothing(mxrow)
+            mxrow = 25
         end
-        write(io, "</p>")
-    end
-    write(io, "<table class=\"data-frame\">")
-    write(io, "<thead>")
-    write(io, "<tr>")
-    write(io, "<th></th>")
-    for column_name in cnames
-        write(io, "<th>$(html_escape(String(column_name)))</th>")
-    end
-    write(io, "</tr>")
-    if eltypes
-        write(io, "<tr>")
-        write(io, "<th></th>")
-        # We put a longer string for the type into the title argument of the <th> element,
-        # which the users can hover over. The limit of 256 characters is arbitrary, but
-        # we want some maximum limit, since the types can sometimes get really-really long.
-        types = Any[eltype(df[!, idx]) for idx in 1:mxcol]
-        ct, ct_title = batch_compacttype(types, 9), batch_compacttype(types, 256)
-        for j in 1:mxcol
-            s = html_escape(ct[j])
-            title = html_escape(ct_title[j])
-            write(io, "<th title=\"$title\">$s</th>")
+
+        mxcol = tryparse(Int, get(ENV, "DATAFRAMES_COLUMNS", "100"))
+
+        if isnothing(mxcol)
```

Comment:
in what case `mxcol` can be `nothing` here?

Reply:
Same answer here.

Reply:
OK